common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java

```diff
@@ -550,25 +619,30 @@ protected synchronized SecretKey loadOrCreateKey()
         }
 
         if (mSecretKeyFromAndroidKeyStore == null) {
-            // If encountering exception for reading keys, try to generate new keys
-            mKeyPair = generateKeyPairFromAndroidKeyStore();
-
-            // Also generate new secretkey
-            mSecretKeyFromAndroidKeyStore = generateSecretKey();
-            final byte[] keyWrapped = wrap(mSecretKeyFromAndroidKeyStore);
-            writeKeyData(keyWrapped);
+            createKey();
         }
 
         return mSecretKeyFromAndroidKeyStore;
     }
 
+    /**
+     * Generate a new key and save to storage.
+     */
+    private synchronized SecretKey createKey() throws GeneralSecurityException, IOException {
+        mSecretKeyFromAndroidKeyStore = generateSecretKey();
```

Comment:
>mSecretKeyFromAndroidKeyStore [](start = 8, length = 29)

mSecretKeyFromAndroidKeyStore is somewhat misleading - it is not coming from the Android key store... Consider renaming it.